### ISSUE TITLE
Mixed-ANOVA error

### ISSUE THREAD

**Issue by berkbuzcu** [raphaelvallat/pingouin#381]
I am running Mixed-ANOVA model on a data structured same as the example.

```
    550 n_rm = len(rm)
--> 551 n_obs = int(grp_with.count().max())
    552 grandmean = data[dv].mean()
    553 # Calculate sums of squares

ValueError: cannot convert float NaN to integer
```

When I retrace the calculation made here I acquire the max count as an integer no problem. Some internal calculation seems to turn the data into an empty group dataframe.

Any thoughts?

**Comment by raphaelvallat** [comment]
Hi,

Can you please share a minimal example (with data and code) to reproduce the example?

Thanks
Raphael

**Comment by berkbuzcu** [comment]
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>
<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Lenovo/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Lenovo/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link=blue vlink=purple>



type | rating | nid | category
-- | -- | -- | --
A | 3 | 0 | A
B | 3 | 0 | A
C | 4 | 0 | A
D | 4 | 0 | B
E | 3 | 0 | B
F | 5 | 0 | B
G | 4 | 0 | C
H | 3 | 0 | C
J | 3 | 0 | C
A | 2 | 1 | A
B | 3 | 1 | A
C | 4 | 1 | A
D | 4 | 1 | B
E | 8 | 1 | B
F | 6 | 1 | B
G | 4 | 1 | C
H | 7 | 1 | C
J | 9 | 1 | C
</body>
</html>


pg.mixed_anova(data=data, dv='rating', between='category', within='type', subject='nid')

**Comment by raphaelvallat** [comment]
Hi @berkbuzcu,

Apologies about the slow reply. The issue is that your data is ill-defined for a mixed ANOVA. If we convert it to a wide-format representation, this becomes clearly visible:

<img width="469" alt="image" src="https://github.com/raphaelvallat/pingouin/assets/25703792/ca4beefe-9f7a-419b-bdb3-4b8b0a33e057">

You need to have:
- Multiple subjects (`nid`) per group (`category`)
- Non-missing values for all repeated measures (`type`)

In other words, there shouldn't be any NaN in the above screenshot.

Thanks
Raphael
